src/subscriptionModules/eventScannerBased.ts

```diff
@@ -90,24 +90,25 @@ export class EventScannerBased implements ISubscriptionModule{
       if(this.isScanOngoing){
         this.isNewScanRequired = true
         this.logger.info(`new scan queued...`)
-      }else{
+      }
+      else{
         try {
-          await this._scanForTransferEvents()
+          do {
+            this.isScanOngoing = true
+            this.isNewScanRequired = false
+            await this._scanForTransferEvents()
+          } while (this.isNewScanRequired);
```

Comment:
One could think isNewScanRequired will always be false at this point. I think even cleaner would be to have _scanForTransferEvents return a bool that you assign to isNewScanRequired.

Reply:
- about cleanliness I'm afraid returning a boolean standing for `isNewScanRequired` from a function that is called `_scanForTransferEvents` is not more intuitive, I should change the name of the function to `_scanForTransferEventsAndIsNewScanRequired` 
  - Also, `_scanForTransferEvents` cannot know if a new scan is required
  - the whole logic should be different, isn't it ?

- I could also move the `this.isNewScanRequired = false` instruction inside the `_scanForTransferEvents` function to prevent that thought... 
  - but then that function would be dealing with something it is not necessary for it to deal with
  - now the concurrency variable are all handled together insider the funciton `_requestNewScan`, we would loose this

- maybe it's fine for someone to think the variable isNewScanRequired will always be false at this point. It will be true only if a scan is requested but a scan is already ongoing

Reply:
in this way you also know that before entering a scan the `isScanOngoing` is always set to true (lock ensured), and all the "queued" requests will be resolved by the new scan that is going to be triggered => `this.isNewScanRequired = false`

Reply:
added comments to the code